ldms/src/sampler/zfs_zpool/zfs_zpool.c

```diff
@@ -0,0 +1,642 @@
+		return (1);
+        }
+
+        if (nvlist_lookup_nvlist(nvroot,
+				ZPOOL_CONFIG_VDEV_STATS_EX, &nv_ex) != 0) {
+                rc = 6;
+        }
+
+	if (rc == 0) {
+		record_instance = ldms_record_alloc(sampler_base->set,
+							index_store.vdev_rec_idx);
+
+		if (record_instance == NULL) {
+			log_fn(LDMSD_LDEBUG, SAMP": ldms_record_alloc() failed, resizing metric set\n");
+			resize_metric_set();
+			record_instance = ldms_record_alloc(sampler_base->set,
+					index_store.vdev_rec_idx);
+			if (record_instance == NULL)
+				rc = 2;
+		}
+	}
+
+	if (rc == 0) {
+		rc = ldms_list_append_record(sampler_base->set, list_handle,
+						record_instance);
+
+		/* zpoolname    0 */
+		ldms_record_array_set_str(record_instance, zpool_metric_ids[0],
+				pool_name);
+	        /* zpool state  1 */
+		ldms_record_array_set_str(record_instance, zpool_metric_ids[1],
+					zpool_state_to_name((vdev_state_t)vs->vs_state,
+					(vdev_aux_t)vs->vs_aux));
+	        /* total     2 */
+		ldms_record_set_u64(record_instance, zpool_metric_ids[2],
+					vs->vs_space);
+	        /* allocated 3 */
+		ldms_record_set_u64(record_instance, zpool_metric_ids[3],
+					vs->vs_alloc);
+	        /* free     4 */
+		ldms_record_set_u64(record_instance, zpool_metric_ids[4],
+					vs->vs_space - vs->vs_alloc);
+	        /* used     5 */
+		cap = (vs->vs_space == 0) ? 0 : (vs->vs_alloc * 10000 / vs->vs_space)/100;
+		ldms_record_set_u64(record_instance, zpool_metric_ids[5], cap);
+
+
+		/* Here we call the get_pool_scan function to fill the rest of the
+		 * record */
+		rc = get_pool_scan_status(nvroot, pool_name, &record_instance);
+
+	}
+        return (0);
+}
+
+
+/*
+ * recursive stats printer
+ */
+static int get_recursive_stats(stat_printer_f func, nvlist_t *nvroot,
+    const char *pool_name, const char *parent_name, int descend)
+{
+        uint_t c, children;
+        nvlist_t **child;
+        char vdev_name[256];
+        int err;
+
+        err = func(nvroot, pool_name, parent_name);
+        if (err)
+                return (err);
+
+        if (descend && nvlist_lookup_nvlist_array(nvroot, ZPOOL_CONFIG_CHILDREN,
+            &child, &children) == 0) {
+                (void) strlcpy(vdev_name, get_vdev_name(nvroot, parent_name),
+                    sizeof (vdev_name));
+
+                for (c = 0; c < children; c++) {
+                        get_recursive_stats(func, child[c], pool_name,
+                                              vdev_name, descend);
+                }
+        }
+        return (0);
+}
+
+
+static int get_zpool_count(zpool_handle_t *zhp, void *data)
+{
+	int rc = 0;
+
+	if (zhp != NULL) {
+		zpool_list_len++;
+	} else {
+		rc = 1;
+	}
+        return (rc);
+}
+
+static int get_pool_scan_status(nvlist_t *nvroot, const char *pool_name, ldms_mval_t *record_instance)
+{
+	uint_t c;
+	int64_t elapsed;
+	uint64_t examined, pass_exam, paused_time, paused_ts, rate;
+	uint64_t remaining_time;
+	pool_scan_stat_t *ps = NULL;
+	double pct_done;
```

Comment:
Essentially unused

Reply:
yup, cut and paste from zfslib db code. Gone